manifests/params.pp

```diff
@@ -29,9 +29,21 @@
   if $::operatingsystem == 'Ubuntu' {
     if versioncmp($::lsbdistrelease, '8.04') < 1 {
       $init_style = 'debian'
-    } else {
+    } elsif versioncmp($::operatingsystemrelease, '15.04') < 0 {
       $init_style = 'upstart'
     }
+    else {
+      # Introducing fact 'is_systemd' since Ubuntu 15.04 can be systemd OR upstart (defaults to systemd though)
+      # Fact allows us to retrieve the client side value.
+      # We convert to boolean incase some people may have stringify facts turned on or not set in puppet 3.x still.
+      $is_systemd = str2bool($::is_systemd)
```

Comment:
Oh nice! This has been a LONG standing issue with the url install method. I myself have been wanting to write one just hadn't had the time.

This is as close as we've gotten https://github.com/solarkennedy/puppet-consul/pull/118